transport/server.cc

```diff
@@ -1116,6 +1129,11 @@ process_execute_internal(service::client_state& client_state, distributed<cql3::
         service_permit permit, tracing::trace_state_ptr trace_state, bool init_trace, cql3::computed_function_values cached_pk_fn_calls,
         cql3::dialect dialect) {
     cql3::prepared_cache_key_type cache_key(in.read_short_bytes(), dialect);
+
+    std::optional<seastar::lw_shared_ptr<cql3::cql_metadata_id_type>> metadata_id_opt;
+    if (client_state.is_protocol_extension_set(cql_protocol_extension::USE_METADATA_ID)) {
+        metadata_id_opt = make_lw_shared(cql3::cql_metadata_id_type(in.read_short_bytes()));
```

Comment:
No, you do materialize it each time.

Drop the lw_shared_ptr. What's the typical size? If it's 31 or less, just use bytes (which has a small-buffer-optimization for 31 bytes). Otherwise use small_vector with the expected size.